### ISSUE TITLE
Embedding JupyterLab into another site without iFrame

### ISSUE THREAD

**Issue by vnerozin** [jupyterlab/jupyterlab#3491]
Is it possible to embed JupyterLab into another site without iFrame?

**Comment by blink1073** [comment]
Hi @vnerozin, are you looking to embed the entire application or certain pieces?  We have examples of both in [examples](https://github.com/jupyterlab/jupyterlab/tree/master/examples).  If you have your own server, you can launch the notebook server with [cross origin](https://github.com/jupyter/jupyter/issues/79#issuecomment-164268449) enabled and set up the [jupyter page config](https://github.com/jupyterlab/jupyterlab/blob/master/examples/notebook/index.html#L8) to use the appropriate `baseUrl` and `wsUrl` for the notebook server.

**Comment by Melchipaul** [comment]
Good morning everyone.
I would like to embed jupyterlab or zeppelin in my website so that user can put their codes and execute them. Do you think is that possible? Thank you.

**Comment by Melchipaul** [comment]
Thank you saulshanabrook for your answer.
And do you know how to do that? Thank you again.

**Comment by saulshanabrook** [comment]
From Steven's suggestion above, I would guess you should follow that to disable CORS (https://github.com/jupyter/jupyter/issues/79#issuecomment-164268449) and then start a jupyterlab server and try putting it in an iframe and see what happens! (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#Examples)

**Comment by bionicles** [comment]
We really need this feature for our dataset editor site because users need to be able to run algorithms on datasets. Iframe doesn't work well because we need to link the state of the dataset to the notebook. If user edits the dataset with the editor, it should reflect on the notebook; if user edits the dataset with the notebook, it should reflect on the editor. 

TLDR: We need a JupyterLab React component in order link the states of the notebook and the app. Anyone know of an example of this working?

**Comment by ziadrida** [comment]
@bionicles have you been able to embed a notebook within your app?

**Comment by sanjitgroverincedo** [comment]
Hi @bionicles, I have the same question - Are you able to embed notebook a notebook within your app? If yes could you share the steps.

**Comment by cport1** [comment]
Build your own UI and use the API is the best way.

**Comment by bionicles** [comment]
I got it to work, sure, but it's massive download for the client, I used https://github.com/executablebooks/thebe but you could also try https://www.npmjs.com/package/juniper-js/v/0.0.3?activeTab=readme

**Comment by Philipp-Blankenhagen** [comment]
Whats the state of this now?
